rest/encoded_response_writer.go

```diff
@@ -66,17 +78,37 @@ func (w *EncodedResponseWriter) WriteHeader(status int) {
 func (w *EncodedResponseWriter) Write(b []byte) (int, error) {
 	w.sniff(b)
 	if w.gz != nil {
-		return w.gz.Write(b)
-	} else {
-		return w.ResponseWriter.Write(b)
+		n, err := w.gz.Write(b)
+		w.lastBytesWritten += int64(n)
+		w.reportStats(false)
+		return n, err
+	}
+	n, err := w.ResponseWriter.Write(b)
+	w.lastBytesWritten += int64(n)
+	w.reportStats(false)
+	return n, err
+}
+
+// ReportStats reports bytes written GetLastBytesWritten returns the number of bytes written by this response writer. This is not locked, so is only safe to call while no one is calling CountedResponseWriter.Write, usually after the response has been fully written.
+func (w *EncodedResponseWriter) reportStats(updateImmediately bool) {
+	if w.bytesWrittenStat == nil {
+		return
+	}
+	currentTime := time.Now()
+	if !updateImmediately && time.Since(currentTime) < w.statsUpdateInterval {
+		return
 	}
+	w.bytesWrittenStat.Add(w.lastBytesWritten)
+	w.lastBytesWritten = 0
+	w.lastReportTime = currentTime
 }
 
 func (w *EncodedResponseWriter) disableCompression() {
 	w.sniffDone = true
 }
 
 func (w *EncodedResponseWriter) sniff(bytes []byte) {
+	fmt.Println(w)
```

Comment:
Why does this need to be added to sniff?

Reply:
mistake.